training/tf/tfprocess.py

```diff
@@ -16,17 +16,18 @@
 #    You should have received a copy of the GNU General Public License
 #    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
 
-import os
 import numpy as np
-import time
+import os
+import random
 import tensorflow as tf
+import time
 
 def weight_variable(shape):
     """Xavier initialization"""
     stddev = np.sqrt(2.0 / (sum(shape)))
     initial = tf.truncated_normal(shape, stddev=stddev)
     weights = tf.Variable(initial)
-    tf.add_to_collection(tf.GraphKeys.REGULARIZATION_LOSSES, weights)
+    tf.add_to_collection(tf.GraphKeys.WEIGHTS, weights)
```

Comment:
I don't understand why you did this renaming. It's now much less clear that bias_variable is also a WEIGHT yet not included in regularization.

Reply:
Looking at https://www.tensorflow.org/api_docs/python/tf/GraphKeys and https://www.tensorflow.org/api_docs/python/tf/contrib/layers/apply_regularization it seems that the intention is that regularizers take tf.GraphKeys.WEIGHTS as inputs and auto-add the resulting losses to tf.GraphKeys.REGULARIZATION_LOSSES.

In this case it doesn't seem to matter too much (as there's no other regularizers in the graph), but if there were, it would result in the output of those regularizers being fed as input into the l2 regularizer which I suspect would not be the intention.